frontend/tests/controllers-test/mainCtrl.test.js

```diff
@@ -0,0 +1,131 @@
+        utilities = _utilities_;
+        $state = _$state_;
+
+        $scope = $rootScope.$new();
+        createController = function () {
+            return $controller('MainCtrl', {$scope: $scope});
+        };
+        vm = createController();
+    }));
+
+    describe('Global variables', function () {
+        it('has default values', function () {
+            vm = createController();
+            expect(vm.user).toEqual({});
+            expect(vm.challengeList).toEqual([]);
+            expect(vm.isChallenge).toBeTruthy({});
+            expect(vm.isMore).toBeFalsy();
+            expect(vm.redirectUrl).toEqual({});
+        });
+    });
+
+    describe('Unit tests for getChallenge function `challenges/featured/`', function () {
+        var success, successResponse;
+        var errorResponse = 'error';
+
+        beforeEach(function () {
+            utilities.sendRequest = function (parameters) {
+                if (success) {
+                    parameters.callback.onSuccess({
+                        data: successResponse
+                    });
+                } else {
+                    parameters.callback.onError({
+                        data: errorResponse
+                    });
+                }
+            };
+        });
+
+        it('successfully get featured challenge', function () {
+            success = true;
+            successResponse = 'success';
+            vm.getChallenge();
+            expect(vm.challengeList).toEqual(successResponse);
+        });
+    });
+
+    describe('Unit tests for init function `auth/user/`', function () {
+        var success, successResponse, errorResponse;
+
+        beforeEach(function () {
+            spyOn($state, 'go');
+            spyOn(utilities, 'resetStorage');
+            utilities.storeData('userKey', 'encrypted key');
+
+            utilities.sendRequest = function (parameters) {
+                if (success) {
+                    parameters.callback.onSuccess({
+                        data: successResponse,
+                        status: 200
+                    });
+                } else {
+                    parameters.callback.onError({
+                        data: errorResponse,
+                        status: 401
+                    });
+                }
+            };
+        });
+
+        it('check for the authenticated user', function () {
+            success = true;
+            successResponse = {
+                username: 'abc123'
+            };
+            vm.init();
+            expect(vm.user.name).toEqual(successResponse.username);
+        });
+
+        it('init function backend error', function () {
+            success = false;
+            errorResponse = 'error';     
+            vm.init();
+            expect(utilities.resetStorage).toHaveBeenCalled();
+            expect($state.go).toHaveBeenCalledWith("auth.login");
+            expect($rootScope.isAuth).toBeFalsy();
+        });
+    });
+
```

Comment:
Validate isMore too.

Reply:
@Sanji515 , Since you are having check for isMore to be false initially if you load a lengthy description you can check isMore is true too..